src/Prelude/Linear/Internal.hs

```diff
@@ -33,12 +33,18 @@ id x = x
 const :: a #-> b -> a
 const x _ = x
 
+asTypeOf :: a #-> a -> a
+asTypeOf = const
+
 -- | @seq x y@ only forces @x@ to head normal form, therefore is not guaranteed
 -- to consume @x@ when the resulting computation is consumed. Therefore, @seq@
 -- cannot be linear in it's first argument.
 seq :: a -> b #-> b
 seq x = Unsafe.toLinear (Prelude.seq x)
 
+($!) :: (a #-> b) #-> a #-> b
+($!) f a = Unsafe.toLinear2 (Prelude.$!) (forget f) a
```

Comment:
Can we re-implement this? No need for an unsafe call, which can block optimisation, for a one-liner.

Reply:
I agree; but if I didn't know I could :). I'd implement it like this:

```
f $! a = a `seq` f a
```

However, that wouldn't be linear since `a` is used twice. But now I figured something like:

```
($!) :: (a #-> b) #-> a #-> b
($!) f !a = f a
```

works. I'll update the PR.

Reply:
First rule of using `Unsafe` is: look at the original implementation. (which, in this case, doesn't work if you copy paste, because of the peskly unrestricted-let limitation, but yours is a solution)